### PR TITLE
Make NodeLaunch attribute strings case sensitive

### LINKED ISSUE #5370: Node launch types should be case-sensitive and lowercase
As per the spec, NodeLanuch string arguments should be interpreted as case-sensitive and lowercase matching other string arguments to attributes:

https://github.com/microsoft/hlsl-specs/pull/62/files#diff-f13bec8f0222a19e56d6433159f69368dd5b5fb9f7b1e0ef0a87058ced354bf9R85

### PR DESCRIPTION
**Pull request by tcorringham**
NodeLaunch string arguments should be interpreted as case-sensitive and lowercase matching other string arguments to attributes. The exisiting implementation followed an earlier spec which allowed case-insensitive NodeLaunch strings - this change disallows that, and amends all the tests that had strings that were not lower case.

Fixes #5370

### CODE REVIEW

File: tools/clang/test/HLSL/workgraph/node_input_compatibility.hlsl [hunk 1]
```diff
@@ -12,7 +12,7 @@ struct RECORD
 [Shader("node")]
 [NumThreads(1024,1,1)]
 [NodeDispatchGrid(64,1,1)]
-[NodeLaunch("BrOaDcasting")]                         // expected-note  {{Launch type defined here}}
```
Comment (llvm-beanz) [inline tools/clang/test/HLSL/workgraph/node_input_compatibility.hlsl:15]: We should probably have tests that verify that we get an expected error with incorrect capitalizations like what was here originally.

File: tools/clang/lib/Sema/SemaHLSL.cpp [hunk 2]
```diff
@@ -12766,9 +12766,6 @@ static StringRef ValidateAttributeStringArg(Sema &S, const AttributeList &A,
```
Comment (llvm-beanz) [inline tools/clang/lib/Sema/SemaHLSL.cpp:12766]: Can we restore this entire function to the version in `main`? The `isCaseSensitive` flag was added for work graphs, so we should be able to just revert this function.

**Review by llvm-beanz — COMMENTED** [review]
Can we get a

File: tools/clang/lib/Sema/SemaHLSL.cpp (hunk 2 above)
Comment (tcorringham) [reply tools/clang/lib/Sema/SemaHLSL.cpp:12766]: I did exactly that in the first commit - it looks as if the subsequent merge reinstated the isCaseSensitive version without me noticing!
I will sort this on return from vacation...

**Review by llvm-beanz — APPROVED** [review]

**Comment by tex3d** [comment]
Considering the impact on all these existing tests which could indicate impact to people using the preview, should we consider adding a clang FixIt for attribute values with mismatched case?

**Review by tex3d — APPROVED** [review]

**Comment by llvm-beanz** [comment]
A fix-it doesn't really get us much since it can't auto-apply in DXC since that infrastructure is broken.

**Pull request merged**